pkg/scheduler/framework/interface.go

```diff
@@ -160,6 +162,33 @@ func (s *Status) AsError() error {
 	return errors.New(s.Message())
 }
 
+// IsStatusEqual returns whether two Statuses are equal to each other.
+func IsStatusEqual(statusX, statusY *Status) bool {
+	if statusX == statusY {
+		return true
+	}
+	if statusX == nil {
+		return statusY == nil || statusY.code == Success
+	}
+	if statusY == nil {
+		return statusX.code == Success
+	}
```

Comment:
`if x == y` would only fit for x is nil and y is nil, right? (if they're both pointers, they are not equal), so prefer to use the following logic to check all cases that x is nil or y is nil:

```go
if x == nil || y == nil {
    return x.IsSuccess() && y.IsSuccess()
}
```

And then the logic below can handle the cases that both x and y are non-nil.